design/user_types.go

```diff
@@ -38,6 +38,71 @@ var UpdateWorkItemPayload = a.Type("UpdateWorkItemPayload", func() {
 	a.Required("type", "fields", "version")
 })
 
+// UpdateWorkItemJSONAPIPayload defines top level structure from jsonapi specs
+// visit : http://jsonapi.org/format/#document-top-level
+var UpdateWorkItemJSONAPIPayload = a.Type("UpdateWorkItemJSONAPIPayload", func() {
+	a.Attribute("data", WorkItemDataForUpdate)
+	a.Required("data")
+})
+
+// WorkItemDataForUpdate defines how an update payload will look like
+var WorkItemDataForUpdate = a.Type("WorkItemDataForUpdate", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("workitems")
+	})
+	a.Attribute("id", d.String, "ID of the work item which is being updated", func() {
```

Comment:
Just like in https://github.com/almighty/almighty-core/blob/master/design/user_types.go#L25
we do not include the ID in the payload because we do it allow updating the ID, and that it could be retrieved from the REST URL, I am wondering if we should include it here.